include/xsimd/config/xsimd_arch.hpp

```diff
@@ -61,6 +61,46 @@ namespace xsimd
         {
         };
 
+        // requires_alignment() as a struct, required because MSVC doesn't
+        // support directly using constexpr functions for SFINAE
```

Comment:
Following one of yourcomment, I do think it would be simpler to always provide an alignment() method, which defaults to ``0``so that this FINAE is not needed.